app/src/main/java/org/pesmypetcare/mypetcare/services/pet/PetManagerAdapter.java

```diff
@@ -87,21 +87,37 @@ public boolean registerNewPet(User user, Pet pet) {
             try {
                 ServiceLocator.getInstance().getPetManagerClient().createPet(user.getToken(), user.getUsername(),
                     libraryPet);
-            } catch (ExecutionException | InterruptedException e) {
+            } catch (MyPetCareException e) {
                 e.printStackTrace();
             }
 
             Pair<DateTime, Double> entry = pet.getLastWeightInfo();
-            Weight libraryWeight = new Weight(entry.first.toString(), entry.second.intValue());
+            Weight libraryWeight = new Weight(entry.first.toString(), entry.second);
             try {
                 ServiceLocator.getInstance().getPetManagerClient().addFieldCollectionElement(user.getToken(),
-                    user.getUsername(), pet.getName(), PetData.WEIGHTS, libraryWeight.getKey(),
+                        user.getUsername(), pet.getName(), PetData.WEIGHTS, libraryWeight.getKey(),
                     libraryWeight.getBodyAsMap());
-            } catch (ExecutionException | InterruptedException e) {
+            } catch (MyPetCareException e) {
                 e.printStackTrace();
             }
         });
 
+        executorService.shutdown();
+        try {
+            executorService.awaitTermination(5, TimeUnit.MINUTES);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['5' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5650471)